sources/execDynawoAlgorithms.sh

```diff
@@ -118,61 +118,89 @@ algo_CS() {
       ;;
     esac
   done
-  
+
   return ${RETURN_CODE}
 }
 
 algo_MC() {
   setEnv
 
-  # launch margin calculation
-  $DYNAWO_ALGORITHMS_INSTALL_DIR/bin/dynawoAlgorithms --simulationType=MC $@
-  RETURN_CODE=$?
-
+  args=""
+  NBPROCS=1
+  FILTER_TIMELINE=false
   while (($#)); do
   case $1 in
     --directory)
       if [ ! -z "$2" ]; then
   	    if [ -d "$2" ]; then
-          filter_timeline $2
+          FILTER_TIMELINE=true
+          timeline=$2
         fi
       fi
-      break
+      args="$args --directory"
+      shift
+      ;;
+    --nbThreads|-np)
+      NBPROCS=$2
+      shift # past argument
+      shift # past value
       ;;
     *)
+      args="$args $1"
       shift
-      break
       ;;
     esac
   done
 
+  # launch margin calculation
+  mpirun -np $NBPROCS $DYNAWO_ALGORITHMS_INSTALL_DIR/bin/dynawoAlgorithms --simulationType=MC $args
```

Comment:
mpirun might not be available in the default path so you need to do the same mechanism than in envDynawoAlg to determine if we need to use the one in the install dir

Reply:
OK